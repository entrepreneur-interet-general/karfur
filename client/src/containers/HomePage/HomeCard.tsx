import React from "react";
import { colors } from "../../colors";
import styled from "styled-components";
import { FButtonMobile } from "../../components/FigmaUI/FButtonMobile/FButtonMobile";

declare const window: Window;
interface Props {
  text: string;
  defaultText: string;
  buttonTitle: string;
  defaultBoutonTitle: string;
  iconName: string;
  image: string;
  t: (text: string, defaultText: string) => void;
  backgroundColor: string;
  textColor: string;
  onClick: (e: any) => void;
  isDisabled: boolean;
}

const MainContainer = styled.div`
  padding: 30px;
  border-radius: 12px;
  height: 440px;
  background-color: ${(props) => props.backgroundColor};
  margin: 8px;
  text-align: center;
`;

const TextContainer = styled.div`
  color: ${(props) => props.textColor};
  font-size: 22px;
  font-weight: 700;
  margin-top: 40px;
  margin-bottom: 20px;
`;

export const HomeCard = (props: Props) => (
  <MainContainer backgroundColor={props.backgroundColor}>
    <img src={props.image} />
    <TextContainer textColor={props.textColor}>
      {props.t(props.text, props.defaultText)}
    </TextContainer>
    <FButtonMobile
      name="search-outline"
      isDisabled={props.isDisabled}
      fill={colors.noir}
      color={colors.blancSimple}
      onClick={props.onClick}
      t={props.t}
      title={props.buttonTitle}
      defaultTitle={props.defaultBoutonTitle}
    />
  </MainContainer>
);
