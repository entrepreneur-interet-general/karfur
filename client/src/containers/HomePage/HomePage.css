/*
  Theme Name: Regna
  Theme URL: https://bootstrapmade.com/regna-bootstrap-onepage-template/
  Author: BootstrapMade.com
  License: https://bootstrapmade.com/license/
*/
/*--------------------------------------------------------------
# General
--------------------------------------------------------------*/
.homepage {
  /* Prelaoder */
  /* Back to top button */
  /*--------------------------------------------------------------
  # Header
  --------------------------------------------------------------*/
  /*--------------------------------------------------------------
  # Hero Section
  --------------------------------------------------------------*/
  /*--------------------------------------------------------------
  # Navigation Menu
  --------------------------------------------------------------*/
  /* Nav Menu Essentials */
  /* Nav Menu Arrows */
  /* Nav Meu Container */
  /* Nav Meu Styling */
  /* Mobile Nav Toggle */
  /* Mobile Nav Styling */
  /* Mobile Nav body classes */
  /*--------------------------------------------------------------
  # Sections
  --------------------------------------------------------------*/
  /* Sections Header
  --------------------------------*/
  /* About Us Section
  --------------------------------*/
  /* Facts Section
  --------------------------------*/
  /* Services Section
  --------------------------------*/
  /* Call To Action Section
  --------------------------------*/
  /* Portfolio Section
  --------------------------------*/
  /* Team Section
  --------------------------------*/
  /* Contact Section
  --------------------------------*/
  /*--------------------------------------------------------------
  # Footer
  --------------------------------------------------------------*/
  /*--------------------------------------------------------------
  # Search bar
  --------------------------------------------------------------*/
}

.homepage a {
  color: #2dc997;
}

.homepage a:hover, .homepage a:active, .homepage a:focus {
  color: #2dca98;
  outline: none;
  text-decoration: none;
}

.homepage p {
  padding: 0;
  margin: 0 0 30px 0;
}

.homepage h1, .homepage h2, .homepage h3, .homepage h4, .homepage h5, .homepage h6 {
  font-family: "Poppins", sans-serif;
  font-weight: 400;
  margin: 0 0 20px 0;
  padding: 0;
}

.homepage #preloader {
  position: fixed;
  left: 0;
  top: 0;
  z-index: 4;
  width: 100%;
  height: 100%;
  overflow: visible;
}

.homepage .back-to-top {
  position: fixed;
  display: none;
  background: rgba(0, 0, 0, 0.2);
  color: #fff;
  padding: 6px 12px 9px 12px;
  font-size: 16px;
  border-radius: 2px;
  right: 15px;
  bottom: 15px;
  transition: background 0.5s;
}

@media (max-width: 768px) {
  .homepage .back-to-top {
    bottom: 15px;
  }
}

.homepage .back-to-top:focus {
  background: rgba(0, 0, 0, 0.2);
  color: #fff;
  outline: none;
}

.homepage .back-to-top:hover {
  background: #2dc997;
  color: #fff;
}

.homepage #header {
  padding: 30px 0;
  height: 92px;
  position: fixed;
  left: 0;
  top: 0;
  right: 0;
  transition: all 0.5s;
  z-index: 2;
}

.homepage #header #logo {
  float: left;
}

.homepage #header #logo h1 {
  font-size: 36px;
  margin: 0;
  padding: 6px 0;
  line-height: 1;
  font-family: "Poppins", sans-serif;
  font-weight: 700;
  letter-spacing: 3px;
  text-transform: uppercase;
}

.homepage #header #logo h1 a, .homepage #header #logo h1 a:hover {
  color: #fff;
}

.homepage #header #logo img {
  padding: 0;
  margin: 0;
}

@media (max-width: 768px) {
  .homepage #header #logo h1 {
    font-size: 26px;
  }
  .homepage #header #logo img {
    max-height: 40px;
  }
}

.homepage #header.header-fixed {
  background: rgba(52, 59, 64, 0.9);
  padding: 20px 0;
  height: 72px;
  transition: all 0.5s;
}

.homepage #hero {
  width: 100%;
  height: 100vh;
  background: url(../../assets/landing/hero-bg.jpg) top center;
  background-size: cover;
  position: relative;
}

@media (min-width: 1024px) {
  .homepage #hero {
    background-attachment: fixed;
  }
}

.homepage #hero:before {
  content: "";
  background: rgba(0, 0, 0, 0.6);
  position: absolute;
  bottom: 0;
  top: 0;
  left: 0;
  right: 0;
}

.homepage #hero .hero-container {
  position: absolute;
  bottom: 0;
  top: 0;
  left: 0;
  right: 0;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  text-align: center;
}

.homepage #hero h1 {
  margin: 30px 0 10px 0;
  font-size: 48px;
  font-weight: 700;
  line-height: 56px;
  text-transform: uppercase;
  color: #fff;
}

@media (max-width: 768px) {
  .homepage #hero h1 {
    font-size: 28px;
    line-height: 36px;
  }
}

.homepage #hero h2 {
  color: #eee;
  margin-bottom: 50px;
  font-size: 24px;
}

@media (max-width: 768px) {
  .homepage #hero h2 {
    font-size: 18px;
    line-height: 24px;
    margin-bottom: 30px;
  }
}

.homepage #hero p {
  font-family: "Poppins", sans-serif;
  font-weight: 400;
  margin: 0;
  padding: 0;
  color: #eee;
  font-size: 24px;
}

@media (max-width: 768px) {
  .homepage #hero p {
    font-size: 18px;
    line-height: 24px;
  }
}

.homepage #hero .btn-get-started {
  font-family: "Poppins", sans-serif;
  text-transform: uppercase;
  font-weight: 500;
  font-size: 16px;
  letter-spacing: 1px;
  display: inline-block;
  padding: 8px 28px;
  border-radius: 50px;
  transition: 0.5s;
  margin: 10px;
  border: 2px solid #fff;
  color: #fff;
}

.homepage #hero .btn-get-started:hover {
  background: #2dc997;
  border: 2px solid #2dc997;
}

.homepage .nav-menu, .homepage .nav-menu * {
  margin: 0;
  padding: 0;
  list-style: none;
}

.homepage .nav-menu ul {
  position: absolute;
  display: none;
  top: 100%;
  left: 0;
  z-index: 1;
}

.homepage .nav-menu li {
  position: relative;
  white-space: nowrap;
}

.homepage .nav-menu > li {
  float: left;
}

.homepage .nav-menu li:hover > ul,
.homepage .nav-menu li.sfHover > ul {
  display: block;
}

.homepage .nav-menu ul ul {
  top: 0;
  left: 100%;
}

.homepage .nav-menu ul li {
  min-width: 180px;
}

.homepage .sf-arrows .sf-with-ul {
  padding-right: 30px;
}

.homepage .sf-arrows .sf-with-ul:after {
  content: "\f107";
  position: absolute;
  right: 15px;
  font-family: FontAwesome;
  font-style: normal;
  font-weight: normal;
}

.homepage .sf-arrows ul .sf-with-ul:after {
  content: "\f105";
}

.homepage #nav-menu-container {
  float: right;
  margin: 0;
}

@media (max-width: 768px) {
  .homepage #nav-menu-container {
    display: none;
  }
}

.homepage .nav-menu a {
  padding: 0 8px 10px 8px;
  text-decoration: none;
  display: inline-block;
  color: #fff;
  font-family: "Poppins", sans-serif;
  font-weight: 400;
  text-transform: uppercase;
  font-size: 13px;
  outline: none;
}

.homepage .nav-menu > li {
  margin-left: 10px;
}

.homepage .nav-menu > li > a:before {
  content: "";
  position: absolute;
  width: 100%;
  height: 2px;
  bottom: 0;
  left: 0;
  background-color: #2dc997;
  visibility: hidden;
  -webkit-transform: scaleX(0);
  transform: scaleX(0);
  transition: all 0.3s ease-in-out 0s;
}

.homepage .nav-menu a:hover:before, .homepage .nav-menu li:hover > a:before, .homepage .nav-menu .menu-active > a:before {
  visibility: visible;
  -webkit-transform: scaleX(1);
  transform: scaleX(1);
}

.homepage .nav-menu ul {
  margin: 4px 0 0 0;
  border: 1px solid #e7e7e7;
}

.homepage .nav-menu ul li {
  background: #fff;
}

.homepage .nav-menu ul li:first-child {
  border-top: 0;
}

.homepage .nav-menu ul li a {
  padding: 10px;
  color: #333;
  transition: 0.3s;
  display: block;
  font-size: 13px;
  text-transform: none;
}

.homepage .nav-menu ul li a:hover {
  background: #2dc997;
  color: #fff;
}

.homepage .nav-menu ul ul {
  margin: 0;
}

.homepage #mobile-nav-toggle {
  position: fixed;
  right: 0;
  top: 0;
  z-index: 3;
  margin: 20px 20px 0 0;
  border: 0;
  background: none;
  font-size: 24px;
  display: none;
  transition: all 0.4s;
  outline: none;
  cursor: pointer;
}

.homepage #mobile-nav-toggle i {
  color: #fff;
}

@media (max-width: 768px) {
  .homepage #mobile-nav-toggle {
    display: inline;
  }
}

.homepage #mobile-nav {
  position: fixed;
  top: 0;
  padding-top: 18px;
  bottom: 0;
  z-index: 2;
  background: rgba(52, 59, 64, 0.9);
  left: -260px;
  width: 260px;
  overflow-y: auto;
  transition: 0.4s;
}

.homepage #mobile-nav ul {
  padding: 0;
  margin: 0;
  list-style: none;
}

.homepage #mobile-nav ul li {
  position: relative;
}

.homepage #mobile-nav ul li a {
  color: #fff;
  font-size: 16px;
  overflow: hidden;
  padding: 10px 22px 10px 15px;
  position: relative;
  text-decoration: none;
  width: 100%;
  display: block;
  outline: none;
}

.homepage #mobile-nav ul li a:hover {
  color: #fff;
}

.homepage #mobile-nav ul li li {
  padding-left: 30px;
}

.homepage #mobile-nav ul .menu-has-children i {
  position: absolute;
  right: 0;
  z-index: 1;
  padding: 15px;
  cursor: pointer;
  color: #fff;
}

.homepage #mobile-nav ul .menu-has-children i.fa-chevron-up {
  color: #2dc997;
}

.homepage #mobile-nav ul .menu-item-active {
  color: #2dc997;
}

.homepage #mobile-body-overly {
  width: 100%;
  height: 100%;
  z-index: 2;
  top: 0;
  left: 0;
  position: fixed;
  background: rgba(52, 59, 64, 0.9);
  display: none;
}

.homepage body.mobile-nav-active {
  overflow: hidden;
}

.homepage body.mobile-nav-active #mobile-nav {
  left: 0;
}

.homepage body.mobile-nav-active #mobile-nav-toggle {
  color: #fff;
}

.homepage .section-header .section-title {
  font-size: 32px;
  color: #111;
  text-transform: uppercase;
  text-align: center;
  font-weight: 700;
  margin-bottom: 5px;
}

.homepage .section-header .section-description {
  text-align: center;
  padding-bottom: 40px;
  color: #999;
}

.homepage #about {
  background: #fff;
  padding: 80px 0;
}

.homepage #about .about-container .background {
  min-height: 300px;
  background: url(../../assets/landing/about-img.jpg) center top no-repeat;
  margin-bottom: 10px;
}

.homepage #about .about-container .content {
  background: #fff;
}

.homepage #about .about-container .title {
  color: #333;
  font-weight: 700;
  font-size: 32px;
}

@media (max-width: 768px) {
  .homepage #about .about-container .title {
    padding-top: 15px;
  }
}

.homepage #about .about-container p {
  line-height: 26px;
}

.homepage #about .about-container p:last-child {
  margin-bottom: 0;
}

.homepage #about .about-container .icon-box {
  background: #fff;
  background-size: cover;
  padding: 0 0 30px 0;
}

.homepage #about .about-container .icon-box .icon {
  float: left;
  background: #fff;
  width: 64px;
  height: 64px;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  text-align: center;
  border-radius: 50%;
  border: 2px solid #2dc997;
}

.homepage #about .about-container .icon-box .icon i {
  color: #2dc997;
  font-size: 24px;
}

.homepage #about .about-container .icon-box .title {
  margin-left: 80px;
  font-weight: 500;
  margin-bottom: 5px;
  font-size: 18px;
  text-transform: uppercase;
}

.homepage #about .about-container .icon-box .title a {
  color: #111;
}

.homepage #about .about-container .icon-box .description {
  margin-left: 80px;
  line-height: 24px;
  font-size: 14px;
}

.homepage #facts {
  background: #f7f7f7;
  padding: 80px 0 60px 0;
}

.homepage #facts .counters span {
  font-size: 48px;
  display: block;
  color: #2dc997;
}

.homepage #facts .counters p {
  padding: 0;
  margin: 0 0 20px 0;
  font-family: "Poppins", sans-serif;
  font-size: 14px;
}

.homepage #services {
  background: #fff;
  background-size: cover;
  padding: 80px 0 60px 0;
}

.homepage #services .box {
  padding: 50px 20px;
  margin-bottom: 50px;
  text-align: center;
  border: 1px solid #e6e6e6;
  height: 200px;
  position: relative;
  background: #fafafa;
}

.homepage #services .icon {
  position: absolute;
  top: -36px;
  left: calc(50% - 36px);
  transition: 0.2s;
  border-radius: 50%;
  border: 6px solid #fff;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  text-align: center;
  width: 72px;
  height: 72px;
  background: #2dc997;
}

.homepage #services .icon a {
  display: inline-block;
}

.homepage #services .icon i {
  color: #fff;
  font-size: 24px;
}

.homepage #services .box:hover .icon {
  background: #fff;
  border: 2px solid #2dc997;
}

.homepage #services .box:hover .icon i {
  color: #2dc997;
}

.homepage #services .box:hover .icon a {
  color: #2dc997;
}

.homepage #services .title {
  font-weight: 700;
  font-size: 18px;
  margin-bottom: 15px;
  text-transform: uppercase;
}

.homepage #services .title a {
  color: #111;
}

.homepage #services .description {
  font-size: 14px;
  line-height: 24px;
}

.homepage #call-to-action {
  background: linear-gradient(rgba(0, 0, 0, 0.6), rgba(0, 0, 0, 0.6)), url(../../assets/landing/call-to-action-bg.jpg) fixed center center;
  background-size: cover;
  padding: 80px 0;
}

.homepage #call-to-action .cta-title {
  color: #fff;
  font-size: 28px;
  font-weight: 700;
}

.homepage #call-to-action .cta-text {
  color: #fff;
}

@media (min-width: 769px) {
  .homepage #call-to-action .cta-btn-container {
    display: flex;
    align-items: center;
    justify-content: flex-end;
  }
}

.homepage #call-to-action .cta-btn {
  font-family: "Poppins", sans-serif;
  text-transform: uppercase;
  font-weight: 500;
  font-size: 16px;
  letter-spacing: 1px;
  display: inline-block;
  padding: 8px 30px;
  border-radius: 50px;
  transition: 0.5s;
  margin: 10px;
  border: 2px solid #fff;
  color: #fff;
}

.homepage #call-to-action .cta-btn:hover {
  background: #2dc997;
  border: 2px solid #2dc997;
}

.homepage #portfolio {
  background: #f7f7f7;
  padding: 80px 0;
}

.homepage #portfolio #portfolio-wrapper {
  padding-right: 15px;
}

.homepage #portfolio #portfolio-flters {
  padding: 0;
  margin: 0 0 45px 0;
  list-style: none;
  text-align: center;
}

.homepage #portfolio #portfolio-flters li {
  cursor: pointer;
  margin: 0 10px;
  display: inline-block;
  padding: 10px 22px;
  font-size: 12px;
  line-height: 20px;
  color: #666666;
  border-radius: 4px;
  text-transform: uppercase;
  background: #fff;
  margin-bottom: 5px;
  transition: all 0.3s ease-in-out;
}

.homepage #portfolio #portfolio-flters li:hover, .homepage #portfolio #portfolio-flters li.filter-active {
  background: #2dc997;
  color: #fff;
}

.homepage #portfolio .portfolio-item {
  position: relative;
  height: 200px;
  overflow: hidden !important;
  margin-bottom: 15px;
  transition: all 350ms ease;
  -webkit-transform: scale(1);
          transform: scale(1);
}

.homepage #portfolio .portfolio-item a {
  display: block;
  margin-right: 15px;
}

.homepage #portfolio .portfolio-item img {
  position: relative;
  top: 0;
  transition: all 600ms cubic-bezier(0.645, 0.045, 0.355, 1);
}

.homepage #portfolio .portfolio-item .details {
  height: 50px;
  background: #2dc997;
  position: absolute;
  width: 100%;
  height: 50px;
  bottom: -50px;
  transition: all 300ms cubic-bezier(0.645, 0.045, 0.355, 1);
}

.homepage #portfolio .portfolio-item .details h4 {
  font-size: 14px;
  font-weight: 700;
  color: #fff;
  padding: 8px 0 2px 8px;
  margin: 0;
}

.homepage #portfolio .portfolio-item .details span {
  display: block;
  color: #fff;
  font-size: 13px;
  padding-left: 8px;
}

.homepage #portfolio .portfolio-item:hover .details {
  bottom: 0;
}

.homepage #portfolio .portfolio-item:hover img {
  top: -30px;
}

.homepage #team {
  background: #fff;
  padding: 80px 0 60px 0;
}

.homepage #team .member {
  text-align: center;
  margin-bottom: 20px;
}

.homepage #team .member .pic {
  margin-bottom: 15px;
  overflow: hidden;
  height: 260px;
}

.homepage #team .member .pic img {
  max-width: 100%;
}

.homepage #team .member h4 {
  font-weight: 700;
  margin-bottom: 2px;
  font-size: 18px;
}

.homepage #team .member span {
  font-style: italic;
  display: block;
  font-size: 13px;
}

.homepage #team .member .social {
  margin-top: 15px;
}

.homepage #team .member .social a {
  color: #b3b3b3;
}

.homepage #team .member .social a:hover {
  color: #2dc997;
}

.homepage #team .member .social i {
  font-size: 18px;
  margin: 0 2px;
}

.homepage #contact {
  background: #f7f7f7;
  padding: 80px 0 40px 0;
}

.homepage #contact #google-map {
  height: 300px;
  margin-bottom: 20px;
}

.homepage #contact .info {
  color: #333333;
}

.homepage #contact .info i {
  font-size: 32px;
  color: #2dc997;
  float: left;
}

.homepage #contact .info p {
  padding: 0 0 10px 50px;
  margin-bottom: 20px;
  line-height: 22px;
  font-size: 14px;
}

.homepage #contact .info .email p {
  padding-top: 5px;
}

.homepage #contact .social-links {
  padding-bottom: 20px;
}

.homepage #contact .social-links a {
  font-size: 18px;
  display: inline-block;
  background: #333;
  color: #fff;
  line-height: 1;
  padding: 8px 0;
  border-radius: 50%;
  text-align: center;
  width: 36px;
  height: 36px;
  transition: 0.3s;
}

.homepage #contact .social-links a:hover {
  background: #2dc997;
  color: #fff;
}

.homepage #contact .form #sendmessage {
  color: #2dc997;
  border: 1px solid #2dc997;
  display: none;
  text-align: center;
  padding: 15px;
  font-weight: 600;
  margin-bottom: 15px;
}

.homepage #contact .form #errormessage {
  color: red;
  display: none;
  border: 1px solid red;
  text-align: center;
  padding: 15px;
  font-weight: 600;
  margin-bottom: 15px;
}

.homepage #contact .form #sendmessage.show, .homepage #contact .form #errormessage.show, .homepage #contact .form .show {
  display: block;
}

.homepage #contact .form .validation {
  color: red;
  display: none;
  margin: 0 0 20px;
  font-weight: 400;
  font-size: 13px;
}

.homepage #contact .form input, .homepage #contact .form textarea {
  border-radius: 0;
  box-shadow: none;
  font-size: 14px;
}

.homepage #contact .form button[type="submit"] {
  background: #2dc997;
  border: 0;
  padding: 10px 24px;
  color: #fff;
  transition: 0.4s;
}

.homepage #contact .form button[type="submit"]:hover {
  background: #51d8ad;
}

.homepage #footer {
  background: #343b40;
  padding: 30px 0;
  color: #fff;
  font-size: 14px;
}

.homepage #footer .copyright {
  text-align: center;
}

.homepage #footer .credits {
  padding-top: 10px;
  text-align: center;
  font-size: 13px;
  color: #ccc;
}

.homepage .input-group.md-form.form-sm.form-2 {
  width: 50%;
}

.homepage .input-group.md-form.form-sm.form-2 .react-autosuggest__container {
  width: 100%;
  position: relative;
  flex: 1 1 auto;
  width: 1%;
}

.homepage .input-group.md-form.form-sm.form-2 .react-autosuggest__container .react-autosuggest__input {
  width: 100%;
  border-top-right-radius: 0;
  border-bottom-left-radius: 0;
}

.homepage .react-autosuggest__container {
  position: relative;
}

.homepage .react-autosuggest__input {
  padding: 10px 20px;
  font-weight: 300;
  font-size: 16px;
  border: 1px solid #aaa;
  border-radius: 4px;
}

.homepage .react-autosuggest__input--focused {
  outline: none;
}

.homepage .react-autosuggest__input--open {
  border-bottom-left-radius: 0;
  border-bottom-right-radius: 0;
}

.homepage .react-autosuggest__suggestions-container {
  display: none;
}

.homepage .react-autosuggest__suggestions-container--open {
  display: block;
  position: absolute;
  top: 51px;
  width: 280px;
  border: 1px solid #aaa;
  background-color: #fff;
  font-family: Helvetica, sans-serif;
  font-weight: 300;
  font-size: 16px;
  border-bottom-left-radius: 4px;
  border-bottom-right-radius: 4px;
  z-index: 2;
}

.homepage .react-autosuggest__suggestions-list {
  margin: 0;
  padding: 0;
  list-style-type: none;
}

.homepage .react-autosuggest__suggestion {
  cursor: pointer;
  padding: 10px 20px;
}

.homepage .react-autosuggest__suggestion--highlighted {
  background-color: #ddd;
}

.homepage .react-autosuggest__section-container {
  border-top: 1px dashed #ccc;
}

.homepage .react-autosuggest__section-container--first {
  border-top: 0;
}

.homepage .react-autosuggest__section-title {
  padding: 10px 0 0 10px;
  font-size: 12px;
  color: #777;
}
