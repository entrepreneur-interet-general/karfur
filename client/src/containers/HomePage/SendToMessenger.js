import React, { PureComponent } from 'react';
import track from 'react-tracking';

class SendToMessenger extends PureComponent {
  componentDidUpdate() {
    const { handleParse } = this.props;
    handleParse();
  }

  render() {
    const {
      color,
      messengerAppId,
      pageId,
      children,
      dataRef,
      size,
    } = this.props;

    return (
      <div
        className="fb-send-to-messenger"
        messenger_app_id={messengerAppId}
        page_id={pageId}
        data-color={color}
        data-size={size}
        data-ref={dataRef}
      >
        {children}
      </div>
    );
  }
}

export default track({
    page: 'SendToMessenger',
  })(SendToMessenger);