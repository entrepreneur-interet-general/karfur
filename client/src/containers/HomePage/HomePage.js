import React, { Component } from 'react';
import track from 'react-tracking';
import { withTranslation } from 'react-i18next';
import { NavLink } from 'react-router-dom';
import { NavHashLink } from 'react-router-hash-link';
import { connect } from 'react-redux';
import { Row, Col, Card, CardHeader, CardBody, CardFooter } from 'reactstrap';
import Swal from 'sweetalert2';
import AnchorLink from 'react-anchor-link-smooth-scroll';

////////A enlever si pas utilisé/////////////:
// import Notifications from '../../components/UI/Notifications/Notifications';
// import SendToMessenger from './SendToMessenger';
// import MessengerSendToMessenger from '../../utils/MessengerSendToMessenger';
import { toggle_lang_modal } from '../../Store/actions/index';
import EVAIcon from '../../components/UI/EVAIcon/EVAIcon';
import FButton from '../../components/FigmaUI/FButton/FButton';
import LanguageBtn from '../../components/FigmaUI/LanguageBtn/LanguageBtn';
import SearchBar from '../UI/SearchBar/SearchBar';
import API from '../../utils/API';

import './HomePage.scss';
import variables from 'scss/colors.scss';

const anchorOffset = '120';

class HomePage extends Component {
  state = {
    search: [{ type: 'Dispositifs', children: [] }, { type: 'Articles', children: [] }, { type: 'Démarches', children: [] }],
    value: '',
    suggestions: [],
  }

  componentDidMount (){
    window.scrollTo(0, 0);
  }

  upcoming = () => Swal.fire( {title: 'Oh non!', text: 'Cette fonctionnalité n\'est pas encore disponible', type: 'error', timer: 1500 })

  render() {
    const { t } = this.props;
    return (
      <div className="animated fadeIn homepage">
        <section id="hero">
          <div className="hero-container">
            <h1>{t("Homepage.Construis ta vie en France", "Construis ta vie en France")}</h1>
            <h5>{t("Homepage.subtitle", {nombre: this.props.dispositifs.length})}</h5>
            
            <div className="search-row">
              <SearchBar 
                validateTest="Valider"
                className="input-group"
                placeholder="Rechercher"
              />
              <div className="try-it-out mr-10">{t("ou", "ou")}</div>
              <FButton tag={NavLink} to="/advanced-search" name="flash" type="dark" className="large-btn">
                {t("Homepage.Super recherche", "Super recherche")}
              </FButton>
            </div>
          </div>
          <AnchorLink offset='60' href="#plan" className="arrowhead-icon header-anchor d-inline-flex justify-content-center align-items-center">
            <EVAIcon className="slide-bottom" name="arrow-circle-down" size="hero"/>
          </AnchorLink>
        </section>

        <section id="plan" className="triptique">
          <div className="section-container">
            <h2>{t("Homepage.Vous cherchez ?", "Tu cherches à ?")}</h2>

            <Row className="card-row">
              <Col lg="4" className="card-col">
                <NavLink to="/advanced-search" className="no-decoration">
                  <Card className="demarche-card">
                    <CardHeader>{t("Homepage.À comprendre une démarche", "À comprendre une démarche")}</CardHeader>
                    <CardBody>
                      {/* <span>Je veux comprendre ce que l'administration me demande et bénéficier de mes droits</span> */}
                    </CardBody>
                    <CardFooter>
                      <FButton type="homebtn" name="search-outline" fill={variables.noir}>
                        {t("Homepage.Trouver une démarche", "Trouver une démarche")}
                      </FButton>
                    </CardFooter>
                  </Card>
                </NavLink>
              </Col>
              <Col lg="4" className="card-col">
                <NavLink to="/advanced-search" className="no-decoration">
                  <Card className="dispo-card">
                    <CardHeader>{t("Homepage.A apprendre", "Rejoindre un dispositif d'accompagnement")}</CardHeader>
                    <CardBody>
                      {/* <span>Je veux rejoindre un dispositif d’accompagnement ou une initiative</span> */}
                    </CardBody>
                    <CardFooter>
                      <FButton type="homebtn" name="search-outline" fill={variables.noir}>
                        {t("Homepage.Trouver un dispositif", "Trouver un dispositif")}
                      </FButton>
                    </CardFooter>
                  </Card>
                </NavLink>
              </Col>
              <Col lg="4" className="card-col">
                <Card className="parcours-card">
                  <CardHeader>{t("Homepage.creer parcours", "Créer ton parcours personnalisé")}</CardHeader>
                  <CardBody>
                    {/* <span>Je veux réaliser mes projets et me construire un avenir qui me plaît</span> */}
                  </CardBody>
                  <CardFooter>
                    <FButton type="homebtn" disabled name="clock-outline" fill={variables.noir}>
                      Bientôt disponible
                    </FButton>
                    {/*<span>{t("Bientôt disponible !", "Bientôt disponible !")}</span>*/}
                  </CardFooter>
                </Card>
              </Col>
            </Row>
          </div>
        </section>

        <section id="contribution" className="contrib">
          <div className="section-container half-width">
            <div className="section-body">
              <h2>{t("Homepage.contributive")}</h2>
              <p className="texte-normal">
                {t("Homepage.contributive subheader")}
                <NavLink className="link" to="/qui-sommes-nous">
                  <p>{t("En savoir plus", "En savoir plus")}</p>
                </NavLink>
              </p>
            </div>
            <footer className="footer-section">
              {t("Homepage.contributeurs mobilises", {nombre: 230})}
              <FButton tag={NavHashLink} to="/comment-contribuer#ecrire" type="dark" className="ml-10">
                {t("Homepage.Je contribue", "Je contribue")}
              </FButton>
            </footer>
          </div>
        </section>

        <section id="multilangues">
          <div className="section-container half-width right-side">
            <div className="section-body">
              <h2>{t("Homepage.disponible langues")}</h2>
              <p className="texte-normal">{t("Homepage.disponible langues subheader")}</p>
              {/*<LanguageBtn />*/}
            </div>
            <footer className="footer-section">
              {t("Homepage.traducteurs mobilises", {nombre: 32})}
              <FButton tag={NavHashLink} to={API.isAuth() ? "/backend/user-profile" : "/comment-contribuer#traduire"} type="dark" className="ml-10">
                {t("Homepage.Je traduis", "Je traduis")}
              </FButton>
            </footer>
          </div>
        </section>

        <section id="certifiee">
          <div className="section-container half-width">
            <div className="section-body">
              <h2>{t("Homepage.information vérifiée")}</h2>
              <p className="texte-normal">{t("Homepage.information vérifiée subheader")}</p>
            </div>
            <footer>
              <FButton tag={NavHashLink} to="/comment-contribuer#corriger" type="dark" className="ml-10">
                {t("En savoir plus", "En savoir plus")}
              </FButton>
            </footer>
          </div>
        </section>

        <section id="explique">
          <div className="section-container half-width right-side">
            <h2>{t("Homepage.Explique les mots difficiles", "Explique les mots difficiles")}</h2>
            <p className="texte-normal">{t("Homepage.explication lexique")}</p>
            <span className="texte-normal">{t("Bientôt disponible !", "Bientôt disponible !")}</span>
            {/*<FButton type="dark" onClick={this.upcoming}>
              Voir le lexique
                </FButton>*/}
          </div>
        </section>
        {/* <div>
            <button onClick={() => this.changeLanguage('fr')}>fr</button>
            <button onClick={() => this.changeLanguage('en')}>en</button>
            <button onClick={() => this.changeLanguage('ar')}>ar</button>
            <h1>{t('Bienvenue')}</h1>
        </div>
        <div>Toolbar, SideDrawer and Backdrop</div>

        <SendToMessenger messengerAppId="300548213983436" pageId="423112408432299" />

        <MessengerSendToMessenger 
          pageId="423112408432299" 
          appId="300548213983436" 
          ctaText="SEND_THIS_TO_ME"
          language= 'fr_FR'
          size="xlarge"
          dataRef="/dispositif/5cd43ce8b472e46bd90e8f58" />

        <Notifications/>

        <div>{t('Elément principal')}</div>
        <div>{t('Elément secondaire')}</div>
        <div>{t('Troisième élément')}</div>
        <div>{t('Quatrième élément')}</div>
        <div>{t('accueil.Cinquième élément')}</div>
        <div>{t('accueil.Six')}</div> */}
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    languei18nCode: state.langue.languei18nCode,
    dispositifs: state.dispositif.dispositifs,
  }
}

const mapDispatchToProps = { toggle_lang_modal };

export default track({
  page: 'HomePage',
})(
  connect(mapStateToProps, mapDispatchToProps)(
    withTranslation()(HomePage)
  )
);