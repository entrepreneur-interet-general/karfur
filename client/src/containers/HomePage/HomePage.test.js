import React from 'react';
import HomePage from './HomePage';
import { expect } from 'chai';
import { shallow } from 'enzyme';
import ReactDOM from 'react-dom';

describe('HomePage', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<HomePage />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should match its reference snapshot', () => {
    const wrapper = shallow(<HomePage />)
    expect(wrapper).to.matchSnapshot()
  });
})
