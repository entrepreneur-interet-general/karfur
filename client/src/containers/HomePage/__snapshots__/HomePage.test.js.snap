// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage should match its reference snapshot 1`] = `
<Connect(withI18nextTranslation(HomePage))
  tracking={
    Object {
      "getTrackingData": [Function],
      "trackEvent": [Function],
    }
  }
/>
`;
